geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/DescriptiveStats.scala

```diff
@@ -0,0 +1,358 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.utils.stats
+
+import org.ejml.data.DenseMatrix64F
+import org.ejml.ops.CommonOps
+import org.ejml.simple.SimpleMatrix
+import org.locationtech.geomesa.utils.stats.SimpleMatrixUtils._
+import org.opengis.feature.simple.SimpleFeature
+
+import scala.collection.JavaConverters._
+import scala.collection.immutable.ListMap
+import scala.Array._
+
+class DescriptiveStats(val attributes: Seq[Int]) extends Stat with Serializable {
+
+  override type S = DescriptiveStats
+
+  private[stats] val size = attributes.size
+  private[stats] val size_squared = size * size
+
+  private[stats] var _count: Long = _
+  private[stats] val _min: SimpleMatrix = new SimpleMatrix(size, 1)
+  private[stats] val _max: SimpleMatrix = new SimpleMatrix(size, 1)
+  private[stats] val _sum: SimpleMatrix = new SimpleMatrix(size, 1)
+  private[stats] val _mean: SimpleMatrix = new SimpleMatrix(size, 1)
+  private[stats] val _m2n: SimpleMatrix = new SimpleMatrix(size, 1)
+  private[stats] val _m3n: SimpleMatrix = new SimpleMatrix(size, 1)
+  private[stats] val _m4n: SimpleMatrix = new SimpleMatrix(size, 1)
+  private[stats] val _c2: SimpleMatrix = new SimpleMatrix(size, size)
```

Comment:
Previous implementations had accessors and `_` was used to differentiate `var` from accessors, this might not be the case any longer...  can adjust if true